ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetriever.java

```diff
@@ -212,14 +230,22 @@ public synchronized void addSidecar(DataColumnSidecar sidecar) {
         existingSidecarsByColIdx.put(sidecar.getIndex(), sidecar);
         if (existingSidecarsByColIdx.size() >= recoverColumnCount) {
           // TODO: Make it asynchronously as it's heavy CPU operation
-          recover();
-          recoveryComplete();
+          try (MetricsHistogram.Timer ignored =
+              dataAvailabilityReconstructionTimeSeconds.startTimer()) {
+            recover();
+            recoveryComplete();
+          } catch (final Throwable t) {
+            LOG.error(
+                "Failed to reconstruct data column sidecar {}", sidecar::toLogString, () -> t);
```

Comment:
It would be not very correct. We already have this sidecar, we want to reconstruct others. Let's log something like 
```java
                "Failed to reconstruct data column sidecars {}", sidecar.getSlotAndBlockRoot(), t);

```